src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/custom.py

```diff
@@ -0,0 +1,168 @@
+
+
+def list_vm(client, lab_name, resource_group, order_by=None, top=None,
+            filters=None, my_vms=None, my_claimable_vms=None):
+    """
+    List virtual machines
+    :param client: Virtual machine client
+    :param lab_name: Name of the lab
+    :param resource_group: Name of the lab resource group
+    :param order_by: The ordering expression for the results, using OData notation
+    :param top: The maximum number of resources to return from the operation
+    :param filters: The filter to apply on the operation
+    :param my_vms: List of viewable labs virtual machines. Cannot be used with --filters
+    :param my_claimable_vms: List of claimable labs virtual machines. Cannot be used with --filters
+    :return:
+    """
+
+    collection = [filters, my_vms, my_claimable_vms]
+    if not _single(collection):
+        raise CLIError("usage error: [--filters FILTER | --my-vms | --my-claimable-vms]")
+
+    if my_vms:
+        # Find out owner object id
+        from azure.cli.core._profile import Profile, CLOUD
+        profile = Profile()
+        cred, _, tenant_id = profile.get_login_credentials(
+            resource=CLOUD.endpoints.active_directory_graph_resource_id)
+        graph_client = GraphRbacManagementClient(cred,
+                                                 tenant_id,
+                                                 base_url=CLOUD.endpoints.active_directory_graph_resource_id)
+        subscription = profile.get_subscription()
+        object_id = _get_object_id(graph_client, subscription=subscription)
+        odata_filter = "Properties/ownerObjectId eq '{}'".format(object_id)
+    elif my_claimable_vms:
+        odata_filter = 'properties/allowClaim'
+    else:
+        odata_filter = filters
+
+    return client.list(resource_group, lab_name, odata_filter, top, order_by)
+
+# TODO: Following methods can be extracted into common utils shared by other command modules
+
+
+def _single(collection):
+    return len([x for x in collection if x]) == 1
+
+
+def _get_object_id(graph_client, subscription=None, spn=None, upn=None):
+    if spn:
+        return _get_object_id_by_spn(graph_client, spn)
+    if upn:
+        return _get_object_id_by_upn(graph_client, upn)
+    return _get_object_id_from_subscription(graph_client, subscription)
+
+
+def _get_object_id_from_subscription(graph_client, subscription):
+    if subscription['user']:
+        if subscription['user']['type'] == 'user':
+            return _get_object_id_by_upn(graph_client, subscription['user']['name'])
+        elif subscription['user']['type'] == 'servicePrincipal':
+            return _get_object_id_by_spn(graph_client, subscription['user']['name'])
+        else:
+            logger.warning("Unknown user type '%s'", subscription['user']['type'])
+    else:
+        logger.warning('Current credentials are not from a user or service principal. '
+                       'Azure Key Vault does not work with certificate credentials.')
```

Comment:
I did straight copy assuming that we'd extract them to common place. What do you think. Should we duplicate or extract as it's only 2 place for now and core might not be the right place to common out :)